src/dune_rules/utop.ml

```diff
@@ -84,47 +85,53 @@ let libs_and_ppx_under_dir sctx ~db ~dir =
                           pps )
                   else (acc, pps))
               | Dune_file.Executables exes -> (
-                let+ libs =
-                  let open Memo.O in
-                  let* compile_info =
-                    let* scope = Scope.DB.find_by_dir dir in
-                    let dune_version =
-                      let project = Scope.project scope in
-                      Dune_project.dune_version project
-                    in
-                    let+ pps =
-                      Resolve.Memo.read_memo
-                        (Preprocess.Per_module.with_instrumentation
-                           exes.buildable.preprocess
-                           ~instrumentation_backend:
-                             (Lib.DB.instrumentation_backend (Scope.libs scope)))
-                      >>| Preprocess.Per_module.pps
+                match Source_tree.Dir.status source_dir with
+                | Vendored | Data_only -> Memo.return (acc, pps)
```

Comment:
Why should we skip vendored libraries?

You can have libraries defined in data_only directories using `subdir`.

Reply:
Isn't this only for executables?

Reply:
It's still suspicious. Anything in the vendored directory is never filtered by `-p`, so the behavior in that issue was expected.